### PR TITLE
lint: update staticcheck to 2021.1.1

### PR DESCRIPTION
**Pull request by stevendanna**
This version contains performance improvements according to the
upstream changelog.

The biggest downside is that the whole-program mode has been removed
from the unused linter. For information about why it was removed see

https://staticcheck.io/changes/2020.2/#unused-whole-program

See individual commits for fixes to new violations added by the new version.

### CODE REVIEW

**Comment by stevendanna** [comment]
Not very scientific, but on this branch it ran a bit faster than the last run on master.

Master:
Screenshot 2021-11-02 at 10 40 00

Branch:
Screenshot 2021-11-02 at 10 39 26

**Comment by stevendanna** [comment]
I don't feel particularly strongly about this. I was hoping the speedup would be a bit more dramatic.

**Review by RaduBerinde — COMMENTED** [review]
:lgtm:

Thanks for this! We should upgrade regardless - it will make it easier to stay up to date in the future.

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/72313#-:-MnZRfiGBHAjadQvCDbD:bzbus7i)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @dt)

File: pkg/geo/geomfn/angle.go [hunk 1]
```diff
@@ -110,7 +110,8 @@ func angleFromCoords(c1, c2, c3, c4 geom.Coord) *float64 {
 	// zero. However, since IEEE 754 requires that -0.0 == 0.0,
 	// this code is still correct and we use -0.0 here for
 	// semantic clarity.
-	if angle == -0.0 { //lint:ignore SA4026
+	//lint:ignore SA4026 -0.0 used here for clarity
+	if angle == -0.0 {
```
Comment (tbg) [inline pkg/geo/geomfn/angle.go:114]: Please actually fix up the other commit with this change.

**Review by tbg — APPROVED** [review]
LGTM

I also think this should be merged.

File: pkg/geo/geomfn/angle.go (hunk 1 above)
Comment (stevendanna) [reply pkg/geo/geomfn/angle.go:114]: Thanks for the reminder!

**Comment by stevendanna** [comment]
We'll need to hold off merging this for a minute. It currently likely causes TestRoachVet to fail on macOS. It appears that one of the updates we pulled in means that we now see the build warning described in https://github.com/cockroachdb/cockroach/issues/70819 as part of the roachvet output. Unfortunately, that means that RoachVet fails since it assumes any output is a failure.

**Comment by stevendanna** [comment]
https://github.com/cockroachdb/cockroach/pull/72686 should resolve the warning

**Review by srosenberg — COMMENTED** [review]
> Reviewed 2 of 2 files at r1, 34 of 34 files at r21, 3 of 3 files at r22, 1 of 1 files at r23, 1 of 1 files at r24, 20 of 20 files at r25, 1 of 1 files at r26, 1 of 1 files at r27, 1 of 1 files at r28.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/72313)* status: :shipit: complete! 0 of 0 LGTMs obtained (and 1 stale) (waiting on @dt, @srosenberg, @stevendanna, and @tbg)

___
*[pkg/spanconfig/spanconfigkvsubscriber/kvsubscriber.go, line 423 at r29](https://reviewable.io/reviews/cockroachdb/cockroach/72313#-MoJqVeR0TP6U0I6sQfF:-MoJqVeR0TP6U0I6sQfG:by3qruz) ([raw file](https://github.com/cockroachdb/cockroach/blob/854ed53ddf4cb9dbcd8d70a258c8987893bc77b7/pkg/spanconfig/spanconfigkvsubscriber/kvsubscriber.go#L423)):*
> ```Go
> }
> 
> func (h *handler) invoke(update roachpb.Span) {
> ```

Another bug...none of the below updates to `h` were previously persisted, so why wasn't anything downstream affected?
___
*[pkg/sql/inverted/expression_test.go, line 84 at r27](https://reviewable.io/reviews/cockroachdb/cockroach/72313#-MoJpw8cDgAO_R6rZ3S-:-MoJpw8cDgAO_R6rZ3S0:b28at99) ([raw file](https://github.com/cockroachdb/cockroach/blob/196f6b37e694ca45af958b64a03590f2b0de5b4f/pkg/sql/inverted/expression_test.go#L84)):*
> ```Go
> 
> func (u *UnknownExpression) IsTight() bool { return u.tight }
> func (u *UnknownExpression) SetNotTight()  { u.tight = false }
> ```

Surprised nothing else was broken considering the previous setter was effectively no-op?

**Review by srosenberg — APPROVED** [review]
:lgtm:

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/72313#-:-MoJqzHN64FWQseXCUU8:bnfp4nl)* status: :shipit: complete! 1 of 0 LGTMs obtained (and 1 stale) (waiting on @dt, @srosenberg, @stevendanna, and @tbg)

**Comment by stevendanna** [comment]
bors r=srosenberg,RaduBerinde,tbg

**Pull request merged**